parse.js

```diff
@@ -609,6 +621,47 @@ var parse = function (buf) {
     })
   })
 
+  schema.messages.forEach(function (msg) {
+    msg.fields.forEach(function (field) {
+      if (field.options && field.options.packed === 'true') {
+        if (PACKABLE_TYPES.indexOf(field.type) === -1) {
+          // let's see if it's an enum
+          if (field.type.indexOf('.') === -1) {
+            if (msg.enums && msg.enums.some(function (en) {
+              return en.name === field.type
+            })) {
+              return
+            }
+          } else {
+            var fieldSplit = field.type.split('.')
+            if (fieldSplit.length > 2) {
+              throw new Error('what is this?')
+            }
+
+            var messageName = fieldSplit[0]
+            var enumName = fieldSplit[1]
+
+            var message
+            schema.messages.some(function (msg) {
+              if (msg.name === messageName) {
+                message = msg
+                return msg
+              }
+            })
+
+            if (message && message.enums && message.enums.some(function (en) {
+              return en.name === enumName
+            })) {
+              return
+            }
+          }
+
+          throw new Error('Fields of type ' + field.type + ' cannot be declared [packed=true]. Only repeated fields of primitive numeric types (types which use the varint, 32-bit, or 64-bit wire types) can be declared "packed". See https://developers.google.com/protocol-buffers/docs/encoding#optional')
```

Comment:
could you make this line a shorter by splitting up the string into multiple lines?

``` js
throw new Error(
  'Field of type' .... +
  '...'
)
```